src/Curator.sol

```diff
@@ -103,16 +118,42 @@ contract Curator is ICurator, UUPS, Ownable, CuratorStorageV1 {
     }
 
     function addListing(address _listing) external {
+        if (isFrozen) revert CURATION_FROZEN();
+
+        if (numAdded - numRemoved == curationLimit) revert CURATION_LIMIT_EXCEEDED();        
+
+        if (msg.sender == owner()) {
+            _addListing(_listing);
+            return;
+        }
+
         if (isPaused) revert CURATION_PAUSED();
 
         if (tokenPass.balanceOf(msg.sender) == 0) revert PASS_REQUIRED();
 
         _addListing(_listing);
     }
 
-    function ownerAddListing(address _listing) external onlyOwner {
-        _addListing(_listing);
-    }
+    function batchAddListing(address[] _listings) external {
+        if (isFrozen) revert CURATION_FROZEN();        
```

Comment:
big fan of just always passing in an array for the listings rather than needing the seperate batch calls, will def add. does the solution u described not run into the issue of having an uncapped dynamic array which gets more expensive to add + remove from as length increaes? i think this is why @kulkarohan moved to this approach in the first place